### PR TITLE
NEW: add shannon entropy

### PR DESCRIPTION
**Pull request by sciunto**
## Description

This PR implements a new feature, the shannon entropy. It's very simple, but we often miss simple features.
This also reflects a personal need for a library I'm writting.

* a request: https://stackoverflow.com/questions/40427172/getting-entropy-of-image-in-python-scikit-image
* wikipedia: https://en.wikipedia.org/wiki/Shannon_(unit)
* matlab page: http://fr.mathworks.com/help/images/ref/entropy.html

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example http://scikit-image.org/docs/dev/auto_examples/filters/plot_entropy.html
- [x] Unit tests

### CODE REVIEW

File: skimage/measure/entropy.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    .. [1] https://en.wikipedia.org/wiki/Entropy_(information_theory)
+    .. [2] https://en.wiktionary.org/wiki/Shannon_entropy
+
+    Examples
+    --------
+    >>> from skimage import data
+    >>> shannon_entropy(data.camera())
+    12.290907503512717
+    """
+    return scipy_entropy(img.ravel())
```
Comment (soupault) [inline skimage/measure/entropy.py]: Could you check if the input image is non-grayscale?

Comment (sciunto) [inline skimage/measure/entropy.py]: done :)

File: skimage/measure/entropy.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    .. [1] https://en.wikipedia.org/wiki/Entropy_(information_theory)
+    .. [2] https://en.wiktionary.org/wiki/Shannon_entropy
+
+    Examples
+    --------
+    >>> from skimage import data
+    >>> shannon_entropy(data.camera())
+    12.290907503512717
+    """
+    assert_nD(image, 2)
```
Comment (jni) [inline skimage/measure/entropy.py]: Please remove this check. The entropy is equally defined for 3D images.

File: skimage/measure/entropy.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    return scipy_entropy(image.ravel())
```
Comment (jni) [inline skimage/measure/entropy.py]: I have a slight preference towards using base 2... Do you have a preference for base e?

File: skimage/measure/entropy.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+from scipy.stats import entropy as scipy_entropy
+from .._shared.utils import assert_nD
+
+
+def shannon_entropy(image):
```
Comment (jni) [inline skimage/measure/entropy.py]: Another slight preference would be for the name to just be `entropy`. At which point we are just a very thin wrapper around scipy, but that's ok.

**Comment by sciunto** [comment]
@jni I believe I addressed your comment. From your feedback I chose to pass the base in the signature, defaulted to bits.

**Comment by sciunto** [comment]
I have to bump scipy to 0.14 (published in Apr 14). I guess it's fine, we depend on numpy 1.11 published in March 16.

File: skimage/measure/entropy.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+from scipy.stats import entropy as scipy_entropy
+from .._shared.utils import assert_nD
+
+
+def shannon_entropy(image, base=2):
+    """
+    Calculate the Shannon entropy of an image.
+
+    The Shannon entropy is defined as S = -sum(pk * log(pk)),
+    where pk are the number of pixels of value k.
```
Comment (emmanuelle) [inline skimage/measure/entropy.py]: With this definition, it's not obvious to me how the entropy is computed on an image of floats. However, I just checked that the entropy is the same whether the image is made of ints or floats.

File: skimage/measure/entropy.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    base : float, optional
+        The logarithmic base to use.
+
+    Returns
+    -------
+    entropy : float
+
+    Notes
+    -----
+    The units are bits or shannon (Sh) for base=2, natural unit (nat) for
```
Comment (emmanuelle) [inline skimage/measure/entropy.py]: I don't understand "units" here.

**Comment by emmanuelle** [comment]
As for the version bump for scipy, is it a runtime only requirement? Can we keep 0.10 in DEPENDS.txt (used for Debian/Ubuntu packaging??) for build requirements and have 0.14 for runtime requirement? @yarikoptic

**Comment by jni** [comment]
@emmanuelle @sciunto I agree that it's tricky to bump up the SciPy requirement. There's been some strong resistance to this in the past. I'm happy with the current changes, but we should perhaps wait to merge this until 0.13 is out. @sciunto do you have a strong desire for this to be in 0.13 or are you ok with waiting one release?

**Comment by sciunto** [comment]
OK.

**Comment by sciunto** [comment]
@jni As the dependencies have been updated. What's the status of this?

File: skimage/measure/entropy.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
+from scipy.stats import entropy as scipy_entropy
+from .._shared.utils import assert_nD
+
+
+def shannon_entropy(image, base=2):
+    """
+    Calculate the Shannon entropy of an image.
```
Comment (jni) [inline skimage/measure/entropy.py]: PEP257 -- Move one-liner to be on the same line as triple quotes

File: skimage/measure/entropy.py (hunk 6 above)
Comment (jni) [inline skimage/measure/entropy.py]: @sciunto maybe clarify, "The returned value is measured in bits..."

**Comment by sciunto** [comment]
Fixed :)

**Pull request merged**